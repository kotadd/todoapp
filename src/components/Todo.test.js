import React from 'react';
import { shallow } from 'enzyme';
import Todo from './Todo';

it('should render Todo component', () => {
  const mockTodo = {
    id: 1,
    text: 'mock todo',
    completed: false
  };

  const wrapper = shallow(<Todo {...mockTodo} />);

  expect(wrapper).toMatchSnapshot();
});
