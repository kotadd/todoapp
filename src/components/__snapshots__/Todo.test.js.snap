// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Todo component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Todo
    completed={false}
    id={1}
    text="mock todo"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "mock todo",
      "onClick": undefined,
      "style": Object {
        "textDecoration": "none",
      },
    },
    "ref": null,
    "rendered": "mock todo",
    "type": "li",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "mock todo",
        "onClick": undefined,
        "style": Object {
          "textDecoration": "none",
        },
      },
      "ref": null,
      "rendered": "mock todo",
      "type": "li",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
}
`;
